client/src/js/modules/fault/views/filters.js

```diff
@@ -27,19 +27,28 @@ define(['marionette',
         },
 
         updateUrlFragment: function() {
-            var frags = {
+            const frags = {
                 bl: 'bl',
                 sys: 'system',
                 com: 'component',
                 sub: 'subcomponent',
             }
 
-            var url = window.location.pathname
+            let url = window.location.pathname
+
+            let searchFrag = ''
+            const searchMatch = url.match(/\/s\/.*$/)
+            if (searchMatch) {
+                searchFrag = searchMatch[0]
+                url = url.replace(searchFrag, '')
+            }
+
             _.each(frags, function(v, f) {
                 url = url.replace(new RegExp('\\/'+f+'\\/\\w+'), '')
-                if (this.ui[v].val() != 0) url += '/'+f+'/'+this.ui[v].val()
+                if (this.ui[v].val() && this.ui[v].val() != 0) url += '/'+f+'/'+this.ui[v].val()
```

Comment:
```suggestion
                if (!!this.ui[v].val()) url += '/'+f+'/'+this.ui[v].val()
```

```suggestion
                if (this.ui[v].val()) url += '/'+f+'/'+this.ui[v].val()
```

Is there any reason why we need the "!= 0"? As 0 is already falsy in JS

As an added measure of precaution, we can even cast the val to a bool with `!!` before evaluating (that means null, undefined, 0, false will all evaluate to false)